### PR TITLE
support input password when using

### PR DESCRIPTION
**Pull request by nowhszh**
- 将password改为可选项
- 当password参数未使用时使用默认密码处理，当传递时使用传递的密码，当未传递时通过交互的方式获取密码

### CODE REVIEW

File: pyextract.py [hunk 1]
```diff
@@ -115,10 +115,10 @@ def extract_and_chmod(args, file):
         os.makedirs(args.output_path)
 
     cmd = "tar -xzvf " + file + " -C " + args.output_path
-    os.system('echo %s | sudo -S %s' % (args.password[0], cmd))
+    os.system(cmd)
 
     cmd = "sudo chmod -R 777" + " " + args.output_path
-    os.system('echo %s | sudo -S %s' % (args.password[0], cmd))
+    os.system('echo %s | sudo -S %s' % (args.password, cmd))
```
Comment (Junbo-Zheng) [inline pyextract.py]: remove `sudo` here?

File: pyextract.py [hunk 2]
```diff
@@ -186,7 +186,7 @@ def extract_and_chmod(args, file):
 
         cmd = "rm -rf " + args.output_path
         print("%s path exist, remove, cmd %s" % (args.output_path, cmd))
-        os.system('echo %s | sudo -S %s' % (args.password[0], cmd))
+        os.system('echo %s | sudo -S %s' % (args.password, cmd))
```
Comment (Junbo-Zheng) [inline pyextract.py]: ditto

File: pyextract.py [hunk 3]
```diff
@@ -214,4 +214,4 @@ def extract_and_chmod(args, file):
 
     # remove output_path since it has been merge to a new file
     cmd = "sudo rm -rf" + " " + args.output_path
-    os.system('echo %s | sudo -S %s' % (args.password[0], cmd))
+    os.system('echo %s | sudo -S %s' % (args.password, cmd))
```
Comment (Junbo-Zheng) [inline pyextract.py]: ditto

**Comment by Junbo-Zheng** [comment]
@nowhszh please rebase and resolve conflicts, since #11 has been merged.

**Comment by nowhszh** [comment]
@Junbo-Zheng done

**Review by Junbo-Zheng — APPROVED** [review]

**Comment by Junbo-Zheng** [comment]
LGTM

File: pyextract.py [hunk 4]
```diff
@@ -26,6 +26,82 @@ def __init__(self):
...
+            '-p', '--purge_source_file',
+            help=
+            'purge source file if is true',
+            action='store_true',
+            default=False)
+        arg_parser.add_argument(
+            '-F', '--filter_pattern',
+            type=str,
+            default=default_cli_parameters.filter_pattern,
+            help='filter the files to be merged')
```
Comment (Junbo-Zheng) [inline pyextract.py:89]: @nowhszh Don't forget to update README.md for usage at the same time. Thanks in advance.

**Pull request merged**